test/lib/repo.js

```diff
@@ -253,57 +249,65 @@ t.test('fail if cannot figure out repo url', t => {
   t.plan(cases.length)
 
   cases.forEach(pkg => {
-    t.test(pkg, t => {
-      repo.exec([['.', pkg].join(sep)], (err) => {
-        t.match(err, { pkgid: pkg })
-        t.end()
-      })
+    t.test(pkg, async t => {
+      t.rejects(
+        command('repo', [['.', pkg].join(sep)]),
+        { pkgid: pkg }
+      )
     })
   })
 })
 
-t.test('open default package if none specified', t => {
-  flatOptions.where = pkgDirs
-  repo.exec([], (er) => {
-    if (er)
-      throw er
-    t.equal(opened['https://example.com/thispkg'], 1, 'opened expected url', {opened})
-    t.end()
-  })
+t.test('open default package if none specified', async t => {
+  npm.localPrefix = pkgDirs
+  await command('repo', [])
+  t.equal(opened['https://example.com/thispkg'], 1, 'opened expected url', {opened})
 })
 
 t.test('workspaces', t => {
-  flatOptions.where = undefined
   npm.localPrefix = join(pkgDirs, 'workspaces')
 
-  t.test('all workspaces', (t) => {
-    repo.execWorkspaces([], [], (err) => {
-      t.notOk(err)
-      t.match({
-        'https://repo.workspace-a/': 1, // Gets translated to https!
-        'https://github.com/npm/workspace-b': 1,
-      }, opened, 'opened two valid repo urls')
-      t.end()
-    })
+  t.afterEach(() => {
+    npm.config.set('workspaces', null)
+    npm.config.set('workspace', [])
+    npm.config.set('include-workspace-root', false)
   })
 
-  t.test('one workspace', (t) => {
-    repo.execWorkspaces([], ['workspace-a'], (err) => {
-      t.notOk(err)
-      t.match({
-        'https://repo.workspace-a/': 1,
-      }, opened, 'opened one requested repo urls')
-      t.end()
-    })
+  t.test('include workspace root', async (t) => {
+    npm.config.set('workspaces', true)
+    npm.config.set('include-workspace-root', true)
+    await command('repo', [])
+    t.match({
+      'https://github.com/npm/workspaces-test': 1,
+      'https://repo.workspace-a/': 1, // Gets translated to https!
+      'https://github.com/npm/workspace-b': 1,
+    }, opened, 'opened two valid repo urls')
   })
 
-  t.test('invalid workspace', (t) => {
-    repo.execWorkspaces([], ['workspace-x'], (err) => {
-      t.match(err, /No workspaces found/)
-      t.match(err, /workspace-x/)
-      t.match({}, opened, 'opened no repo urls')
-      t.end()
-    })
+  t.test('all workspaces', async (t) => {
+    npm.config.set('workspaces', true)
+    await command('repo', [])
+    t.match({
+      'https://repo.workspace-a/': 1, // Gets translated to https!
+      'https://github.com/npm/workspace-b': 1,
+    }, opened, 'opened two valid repo urls')
+  })
+
+  t.test('one workspace', async (t) => {
+    npm.config.set('workspace', ['workspace-a'])
+    await command('repo', [])
+    t.match({
+      'https://repo.workspace-a/': 1,
+    }, opened, 'opened one requested repo urls')
+  })
+
```

Comment:
```suggestion

  t.test('root workspace', async (t) => {
    npm.config.set('workspace', ['workspaces-test'])
    await command('repo', [])
    t.match({
      'https://github.com/npm/workspaces-test': 1,
    }, opened, 'opened one requested repo urls')
  })

```
I haven't confirmed that this proposed test passes; should it? Specify the package name of the root.

Reply:
No it should not. Referring to the root package as a workspace is not a valid approach.

Reply:
I see, thank you. If I want to do that (specify packages by name, including the root package) may I open a feature request?

Reply:
No, referring to your root package as a workspace is not going to be supported. It was explored as part of this feature and discussed in today's RFC call. It is not a good idea in our opinion.

The `--include-workspace-root` flag is the thing that will enable this behavior.  

See this issue for the full rundown of the options that as far as we can tell covers all the bases https://github.com/npm/statusboard/issues/371

Reply:
Thank you, I've reviewed the [meeting notes](https://github.com/npm/rfcs/blob/latest/meetings/2021-08-18.md#issue-371-featworkspaces-introduce-new-workspaces-root-selector---darcyclarke) and the [video recording](https://youtu.be/nkohPHyvzw4?t=2340). Is it purely about nomenclature and the definition of "workspace" vs. "package", or is there a technical reason why referring to your root package by name isn't a good idea?

https://github.com/npm/statusboard/issues/371 covers all the bases, I'd just like to refer to all packages uniformly by name -- unless there's a reason not to? The same way I `import { a } from "workspace-x"` and `import { b } from "root-name"`.

Reply:
The reason is where you're referring to it.  Referring to it under the context of `workspaces` is the problem point from a technical standpoint.  It is not a workspace and is not treated as one, as I believe Isaac iterated during the call.